### PR TITLE
Remove phpunit and prophecy as requirement

### PR DESCRIPTION
**Pull request by alexander-schranz**

### CODE REVIEW

**Comment by alexander-schranz** [comment]
@localheinz this should not be. see the official plugins:

 - https://github.com/phpstan/phpstan-symfony/blob/master/composer.json
 - https://github.com/phpstan/phpstan-doctrine/blob/master/composer.json

I updated the code

**Comment by alexander-schranz** [comment]
@localheinz done

File: composer.json [hunk 1]
```diff
@@ -10,10 +10,16 @@
...
-        "phpspec/prophecy": "^1.7",
-        "phpunit/phpunit": "^6.0||^7.0",
         "phpstan/phpstan": "^0.10"
     },
+    "require-dev": {
+        "phpspec/prophecy": "^1.7",
+        "phpunit/phpunit": "^6.0||^7.0"
+    },
+    "conflicts": {
+        "phpspec/prophecy": "<1.7",
```
Comment (Jean85) [inline composer.json]: Should we conflict with 2.0 too, to reflect the same behavior of requiring `^1.7`?

**Comment by alexander-schranz** [comment]
@Jean85 done

**Comment by localheinz** [comment]
@alexander-schranz 

This needs a quick rebase following #21, but I think then it is ready to go.

👍

File: composer.json [hunk 2]
```diff
@@ -10,10 +10,16 @@
...
-        "phpunit/phpunit": "^6.0||^7.0",
         "phpstan/phpstan": "^0.10"
     },
+    "require-dev": {
+        "phpspec/prophecy": "^1.7",
+        "phpunit/phpunit": "^6.0||^7.0"
+    },
+    "conflicts": {
+        "phpspec/prophecy": "<1.7, >=2.0",
+        "phpunit/phpunit": "<6.0"
```
Comment (Jean85) [inline composer.json]: Sorry for being pedanting, but maybe we should do the same here, conflicting with `<6, >=8`

**Comment by alexander-schranz** [comment]
anything missing?

**Comment by Jan0707** [comment]
I do not really see the benefit of using class strings over the native `::class` call. But maybe I am missing something here :)

**Comment by alexander-schranz** [comment]
@Jan707 does not fail on global ci check wenn prophecy is not used in current checked project.

**Comment by acasademont** [comment]
Would be great to have this merged, Symfony 4 complains a bit ;)

**Comment by Jan0707** [comment]
Thanks @alexander-schranz  I did not consider global installs ...

**Pull request merged**